packages/datadog-instrumentations/src/restify.js

```diff
@@ -37,18 +40,37 @@ function wrapFn (fn) {
   if (Array.isArray(fn)) return wrapMiddleware(fn)
 
   return function (req, res, next) {
-    if (req.route) {
-      routeChannel.publish({ req, route: req.route })
+    if (typeof next === 'function') {
+      arguments[2] = wrapNext(req, next)
     }
 
-    return fn.apply(this, arguments)
+    const route = req.route && req.route.path
+
+    enterChannel.publish({ req, route })
+
+    try {
+      return fn.apply(this, arguments)
+    } catch (error) {
+      errorChannel.publish({ req, error })
+      nextChannel.publish({ req })
+      exitChannel.publish({ req })
+    }
+  }
+}
+
+function wrapNext (req, next) {
+  return function () {
+    nextChannel.publish({ req })
+    exitChannel.publish({ req })
+
+    next.apply(null, arguments)
```

Comment:
why are we forcing it to `null` instead of use `this`?

Reply:
It was a copy paste from another file that did the same. I don't know why so I'll change it.